### ISSUE TITLE
Crash when using visualizer (sys.excepthook error)

### ISSUE THREAD

**Issue by praeludiumOrbis** [circleguard/circlevis#9]
```
[ERROR] 2021/01/24 20:30:36.0143 sys.excepthook error
Type: <class 'IndexError'>
Value: index 5777 is out of bounds for axis 0 with size 5777
Traceback:   File "site-packages\circlevis\renderer.py", line 290, in paintEvent
  File "site-packages\circlevis\renderer.py", line 316, in paint_cursor

 (circleguard_gui, main.py:129)
NoneType: None
```

I think this happened on https://osu.ppy.sh/b/465524?m=0 but I'm not sure

It happened a couple of times before:

```
[ERROR] 2021/01/24 20:06:53.0869 sys.excepthook error
Type: <class 'IndexError'>
Value: index 4966 is out of bounds for axis 0 with size 4966
Traceback:   File "site-packages\circlevis\renderer.py", line 290, in paintEvent
  File "site-packages\circlevis\renderer.py", line 316, in paint_cursor

 (circleguard_gui, main.py:129)
NoneType: None
```

```
[ERROR] 2021/01/14 12:49:05.0093 sys.excepthook error
Type: <class 'IndexError'>
Value: index 4533 is out of bounds for axis 0 with size 4533
Traceback:   File "site-packages\circlevis\renderer.py", line 290, in paintEvent
  File "site-packages\circlevis\renderer.py", line 316, in paint_cursor

 (circleguard_gui, main.py:129)
NoneType: None
```

**Comment by Liam-DeVoe** [comment]
Not sure what's going on here. Can't reproduce (though I only tried for a few minutes) and there's no obvious cause from the logs. Would appreciate knowing what replay, where in the map, and on what settings (speed, rewinding or normal direction, raw view, num frames, etc) this happens if you catch it in the act again.

**Comment by Liam-DeVoe** [comment]
going to close this as a "I have no idea what the issue was and hopefully it's been fixed since". Will reopen if it happens again

**Comment by theIglooo** [comment]
may have reproduced this @tybug

Steps ([and a video](https://streamable.com/ynxo9c)):
1. Visualize these replays together ([one](https://files.catbox.moe/zupgme.osr), [two](https://files.catbox.moe/fjcl1l.osr))
2. Pause and go to the beginning
3. Press left arrow key twice

The log:

```
 Traceback (most recent call last):
  File "circlevis/renderer.py", line 415, in paintEvent
  File "circlevis/renderer.py", line 450, in paint_cursor
IndexError: index 14937 is out of bounds for axis 0 with size 14937
QBackingStore::endPaint() called with active painter; did you forget to destroy it or call QPainter::end() on it?
QPainter::begin: Painter already active
fish: Job 1, 'dist/circleguard' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by Liam-DeVoe** [comment]
thanks for the repro case. This should be properly fixed now with 35835647939e729e277c0a34290a170ee3cd01fd
